libs/chrome-ext/src/lib/background.js

```diff
@@ -104,6 +123,52 @@ function uploadLink(settings, url, labels) {
     .subscribe();
 }
 
+function getChannelVideos(settings, channelId, labels) {
+  loadPaginated(
+    `https://www.googleapis.com/youtube/v3/search?key=${settings.YOUTUBE_KEY}&channelId=${channelId}&part=snippet,id&order=date&maxResults=50`,
+  )
+    .then((items) =>
+      items
+        .filter((video) => video.id.kind === 'youtube#video')
+        .map((video) => ({
+          id: video.id.videoId,
+          title: video.snippet.title,
+          date: video.snippet.publishedAt,
+        }))
+        .sort((a, b) => b.date.localeCompare(a.date)),
+    )
+    .then((videos) => selectVideos(videos, labels));
+}
+
+function getPlaylistVideos(settings, playlistId, labels) {
+  loadPaginated(
+    `https://www.googleapis.com/youtube/v3/playlistItems?key=${settings.YOUTUBE_KEY}&playlistId=${playlistId}&part=snippet,id&order=date&maxResults=50`,
+  )
+    .then((items) =>
+      items
+        .filter((video) => video.snippet.resourceId.kind === 'youtube#video')
+        .map((video) => ({
+          id: video.snippet.resourceId.videoId,
+          title: video.snippet.title,
+          date: video.snippet.publishedAt,
+        }))
+        .sort((a, b) => b.date.localeCompare(a.date)),
+    )
+    .then((videos) => selectVideos(videos, labels));
+}
+
+function selectVideos(videos, labels) {
+  chrome.storage.local.set({ videos });
+  chrome.tabs.create({ url: `youtube/selection.html?labels=${JSON.stringify(labels)}` });
+}
+
+function uploadLinksList(list, labels) {
+  if (!(list && list.length && list.length > 0)) {
```

Comment:
I find this way of doing the exclusion a bit hard to read, why not using `||` instead of the `&&` so you can make it clearer?
```suggestion
  if (!list || !list.length || list.length === 0) {
```